source/Hardpoint.cpp

```diff
@@ -350,20 +402,31 @@ bool Hardpoint::FireSpecialSystem(Ship &ship, const Body &body, std::vector<Visu
 	// Get the weapon range. Anti-missile and tractor beam shots always last a
 	// single frame, so their range is equal to their velocity.
 	double range = outfit->Velocity();
+	Angle facing = ship.Facing();
 
 	// Check if the body is within range of this hardpoint.
-	Point start = ship.Position() + ship.Facing().Rotate(point);
+	Point start = ship.Position() + facing.Rotate(point);
 	Point offset = body.Position() - start;
 	if(offset.Length() > range)
 		return false;
 
+	// Check if the target is within the arc of fire.
+	Angle aim(offset);
+	if(!IsOmnidirectional())
```

Comment:
When the function just returns a value to which we have direct access, I think it's better to use the variable.
```suggestion
	if(!isOmnidirectional)
```